### PR TITLE
DIV-3956 - initial configuration for divorce alerts

### PR DESCRIPTION
**Pull request by hosainnet**
**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines
- [x] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

DIV-3956

### Change description ###

0) Updated the azurerm version to the latest version
1) Created an action group, which is set up with the divorce support mailing list
2) Created an alert to trigger an email if any 400 HTTP errors have occur 
3) Created an alert to trigger an email if any 5xx HTTP errors occur (excluding /health)
4) Created an alert to trigger an email if any FE page takes longer than 10 seconds to render
5) Added sandbox vars file 

We may have to adjust the values above over time (trigger threshold etc) depending on how reliable the logging is

Reference links:

* https://github.com/hmcts/cmc-shared-infrastructure
* https://github.com/hmcts/cnp-module-action-group
* https://github.com/hmcts/cnp-module-metric-alert

Thanks @alexwakeman for the initial implementation

You can check what the alerts look like in Azure Portal -> div-sandbox (application insights) -> Alerts

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: action-groups.tf [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  source   = "git@github.com:hmcts/cnp-module-action-group"
+  location = "global"
+  env      = "${var.env}"
+
+  resourcegroup_name     = "${azurerm_resource_group.rg.name}"
+  action_group_name      = "div-support"
+  short_name             = "Div Support"
+  email_receiver_name    = "Divorce Support Mailing List"
+  email_receiver_address = "${data.azurerm_key_vault_secret.divorce_support_email_secret.value}"
+}
```
Comment (hosainnet) [inline action-groups.tf]: Action group with an email address setup. The email is retrieved from the vault

File: alerts.tf [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  app_insights_name = "div-${var.env}"
+
+  alert_name = "div-bad-requests"
+  alert_desc = "Found HTTP requests with 400 error response code (bad request) in div-${var.env}."
+  app_insights_query = "requests | where resultCode == \"400\""
+  custom_email_subject = "Alert: bad requests in div-${var.env}"
+  frequency_in_minutes = 5
+  time_window_in_minutes = 5
+  severity_level = "2"
+  action_group_name = "Divroce Support"
```
Comment (qzhou-hmcts) [inline alerts.tf]: Divorce

File: alerts.tf [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  app_insights_name = "div-${var.env}"
+
+  alert_name = "div-server-errors"
+  alert_desc = "Found HTTP requests with 400 error response code (bad request) in div-${var.env}."
+  app_insights_query = "requests | where resultCode startswith \"5\"  | where name !contains \"/health\""
+  custom_email_subject = "Alert: server errors in div-${var.env}"
+  frequency_in_minutes = 5
+  time_window_in_minutes = 5
+  severity_level = "2"
+  action_group_name = "Divroce Support"
```
Comment (qzhou-hmcts) [inline alerts.tf]: Divorce

File: alerts.tf [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+}
+
+module "div-fe-performance-alert" {
+  source = "git@github.com:hmcts/cnp-module-metric-alert"
+  location = "${var.location}"
+
+  app_insights_name = "div-${var.env}"
+
+  alert_name = "div-fe-performance-alert"
+  alert_desc = "Web pages took longer than 10 seconds to load in div-${var.env}."
```
Comment (qzhou-hmcts) [inline alerts.tf]: Not really sure if this alert will be really useful, is frequent having request slower than 10 secs, (12 in last 24h). Azure has alert when the performance has degraded, We received them on aat, probably devops received them from prod

Comment (hosainnet) [reply alerts.tf]: The minimum is 5 requests taking longer than 10 seconds within a 5 minute window. I had a look at the 12 you've mentioned and they're spread around so this wouldn't have triggered for those anyway.

I saw the Azure performance degraded emails you've mentioned, and they seem to refer to particular components.
This is a simple "Is website suddenly slower as a user for a long period of time E2E?" (think CCD slowness) 

We may have to tweak the frequency/threashold overtime if it becomes too noisy, but I think it's very valuable

File: alerts.tf (hunk 3 above)
Comment (hosainnet) [reply alerts.tf]: 👍

File: alerts.tf (hunk 2 above)
Comment (hosainnet) [reply alerts.tf]: 👍

**Pull request merged**